packages/compat/tests/stage2.test.ts

```diff
@@ -657,4 +657,111 @@ describe('stage2 build', function () {
       assertFile.matches(/console\.log\(true\)/);
     });
   });
+
+  describe('engines with css', function () {
+    let build: BuildResult;
+    let expectFile: ExpectFile;
+
+    beforeAll(async function () {
+      let app = Project.emberNew();
+      let buildOptions: Partial<BuildParams> = {
+        stage: 2,
+        type: 'app',
+        emberAppOptions: {
+          tests: false,
+          babel: {
+            plugins: [],
+          },
+        },
+        embroiderOptions: {},
+      };
+
+      app.linkPackage('ember-engines');
+
+      let lazyEngine = app.addEngine('lazy-engine', true);
+      let eagerEngine = app.addEngine('eager-engine', false);
+
+      merge(lazyEngine.files, {
+        addon: {
+          styles: {
+            'addon.css': `.lazy { background-color: red; }`,
+          },
+        },
+      });
+
+      merge(eagerEngine.files, {
+        addon: {
+          styles: {
+            'addon.css': `.eager { background-color: blue; }`,
+          },
+        },
+      });
+
+      build = await BuildResult.build(app, buildOptions);
+      expectFile = expectFilesAt(build.outputPath);
+    });
+
+    afterAll(async function () {
+      await build.cleanup();
+    });
+
+    test('lazy engines appear in _embroiderEngineBundles_', function () {
+      expectFile('assets/my-app.js').matches(`w._embroiderEngineBundles_ = [
+  {
+    names: ["lazy-engine"],
+    load: function() {
+      return import("./_engine_/lazy-engine.js");
+    }
+  },
+]`);
+    });
+
+    test('lazy engine css is imported', function () {
+      expectFile('assets/_engine_/lazy-engine.js')
```

Comment:
So, one way I test css stuff in an acceptance test Is to start the app, and inspect the relevant element's style property. Rather than parsing/matching content in output files. Is this relevant here?

The current approach indirectly tests the system, based both on what appears to be build specific internals but we have little confidence that the css is loaded, and loaded in the right order.

Reply:
Speaking with travis, it seems this PR should add an acceptance test scenario for this case. And (as time permits, or maybe as a follow up) adjust how we test these intermediate stage 2 outputs.

For example, today we are testing that a file contains a strong, but that isn't a very durable mechanism for testing.

Rather, we should execute these files in a carefully constructed environment and ensure they have the side-affects we expect. It seems like these files are not one-off, and having this machinery will enable testing all of them.

One thought would be to (with a modern version of node). Context a vmcontext that is able to execute these files.

https://nodejs.org/api/vm.html#vm_class_vm_module
Given that the vm context now has, although experimental, support for SourceTextModules creation, linking, and evaluation. see: https://nodejs.org/api/vm.html#vm_class_vm_module.

My thought here is, we invest a little in such a harness then we can execute actual tests against them. Ones who do not care about the actual source-code, but only the side-affects.

Only risk I see, is that the node API is currently unstable. If that is a concern, this could be implemented as a code-transform instead, although I would likely recommend falling back to a code-transform only if the node folks fundamentally break this api.

Reply:
This is a misunderstanding of how our tests are structured.

If you want to go end-to-end with actual execution in the browser, use one of the many test packages that run entire ember test suites. Don’t go adding VM execution to these tests, it’s fake and inferior to browser testing anyway. 

Tests line the ones in stage2.test.ts are deliberately testing a static view of the build, not a runtime one.

Reply:
@ef4 the above is two thoughts, does your comment apply to both points or just the vm thought?

Specifically, are you concerned with:

> Speaking with travis, it seems this PR should add an acceptance test scenario for this case [blocker].

Reply:
@ef4 / @stefanpenner Should I add an additional engine that doesn't contain any styles or are we happy with theses tests

Reply:
I would be OK with adding an end-to-end test but don't consider it a blocker.

What I was trying to say in reply to @stefanpenner above is that I most definitely do not want any tests that try to evaluate browser code inside node's vmcontext. Any time someone feels tempted to do that, they should move the test into one of our many ember apps to test in the browser instead. These node tests are deliberately looking at a state view of the output, not a runtime view. It's important to have both views, because there are things you can't prove with only a runtime view (for example -- that dead code isn't present).